src/util/dapps.js

```diff
@@ -128,16 +132,49 @@ export function fetchBuiltinApps (api) {
 }
 
 export function fetchLocalApps (api) {
-  return api.parity.dappsList()
-    .then((apps) => {
-      return apps
-        .map((app) => {
-          app.type = 'local';
-          app.visible = true;
-          return app;
-        })
-        .filter((app) => app.id && !['ui'].includes(app.id));
-    })
+  const dappsPath = getLocalDappsPath();
+
+  return fsReaddir(dappsPath) // List files
+    .then(filenames => // Gather info about files
+      Promise.all(filenames.map(filename => {
+        const filePath = path.join(dappsPath, filename);
+
+        return fsStat(filePath).then(stat => ({ isDirectory: stat.isDirectory(), filePath, filename }));
+      }))
+    )
+    .then(files => // Only keep directories
+      files.filter(({ isDirectory }) => isDirectory))
+    .then(dappsFolders => // Parse manifests
+      Promise.all(dappsFolders.map(({ filePath, filename }) => {
+        const manifestPath = path.join(filePath, 'manifest.json');
+
+        return fsReadFile(manifestPath).then(r => {
+          try {
+            return { filename, manifest: JSON.parse(r) };
+          } catch (e) {
+            console.error(`Couldn't parse manifest.json for local dapp ${filePath}`, e);
+            return { filename };
+          }
+        }).catch(e => {
+          console.error(`Couldn't read manifest.json for ${filePath}`);
+          return { filename };
+        });
+      })))
+    .then(dapps =>
+      dapps.filter(({ manifest }) => manifest))
+    .then(dapps =>
+      dapps.map(({ filename, manifest: { id, localUrl, iconUrl, ...rest } }) => (
+        {
+          ...rest,
+          type: 'local',
+          // Prevent using the appId of an existing non-local dapp with already approved permissions
+          id: `LOCAL-${id}`,
+          folderName: 'XYZ',
+          visible: true,
```

Comment:
I think you can remove `visible: true`, I think it's not used. Anyways, this should not be how we manage visibility of dapps, there's Browse Dapps for that.

Reply:
The local dapps don't show up by default if I remove `visible: true` here (although they still can be displayed manually using `Browse Dapps`).

This only [sets the default visibility](https://github.com/parity-js/shell/blob/2a0a39637f6d53d5112109d5e922f12def85ff3b/src/Dapps/store.js#L349-L353) of the dapp; do you think this is problematic?